### DISCUSSION TITLE
Problem with converting scale color on only one PDF

### DISCUSSION THREAD

**Discussion by MisiekDP** [ImageMagick/ImageMagick#6184] [Help] [answered]
### ImageMagick version

6.9.11-60 Q16 arm 2021-01-25

### Operating system, version and so on

RaspberryPi, Debian

### Description

Hey,
I have encountered a strange PDF file that causes me an error when converting to a different color scale.
In general operation the script works perfectly, it can convert almost any PDF file, however, I have a problem with the following file and it is the only one that causes the error.
Could I get some help as to why this is happening? Perhaps I should change some parameters in the policies.xml file?

Script:
`
convert -density "${valDensity}" "$splitPdfFile" -colorspace sRGB -background white -alpha background -alpha remove +level-colors magenta,white -quality 100 "$toFolder/2_pdf_to_color_scale_pdf_and_ready_to_print/$filenameOfFileWithoutExtension.pdf"`

Error when converting a PDF file:
`mar 21 23:34:01 raspberrypi bash[2235]: convert-im6.q16: cache resources exhausted /tmp/magick-ZUK_IdqaCy43w_ChGJg_wF1MkFPi7ZKa1' @ error/cache.c/OpenPixelCache/4095.
mar 21 23:34:01 raspberrypi bash[2235]: convert-im6.q16: no images defined /home/pi/!SambaShareFolder/!EpsonPrintFromRPI/to/2_pdf_to_color_scale_pdf_and_ready_to_print/A4_200dpi_Dokx — kopia_page_01.pdf' @ error/convert.c/ConvertImageCommand/3229.`

The PDF file which causing an error:
[PDF file.pdf](https://github.com/ImageMagick/ImageMagick/files/11034512/PDF.file.pdf)

**Comment by snibgo** [comment]
What density have you specified? The PDF file has one page, 53 by 73 inches. I suppose your density means IM will try to rasterize too many pixels for your available memory.

Try with a very low density such as "1".

**Reply by MisiekDP** [reply]
Hmm, I have density from 200 too 400. But with 200 it is not possible to convert too. How can I edit this pdf to try it print with good quality with density 300?

**Comment by snibgo** [comment, marked as answer]
The PDF looks like an ordinary document that would normally be about 8x11 inches. But it is 53x72 inches, which is as tall as me. Why do you need that rasterized at 300 dpi? That would be 15900x21600 pixels, 350 million pixels, which is huge.

I can do that with no problem on a 16 GB laptop, but your Raspberry Pi might have less memory.

> Perhaps I should change some parameters in the policies.xml file?

Perhaps. What parameters are set?

What does "`convert -list resource`" say?

**Reply by MisiekDP** [reply]
Ok, it is ok, I changed the document size to smaller. You are right that this document should be A4 size paper.
I changed the size and now everything works with this document too :)
Thanks!
